### PR TITLE
WIP: rating-end-points

### PR DESCRIPTION
**Pull request by raeyh**
Here I tried to model the Ratings and get to fetch all the ratings,
All end points are not implemented,
Add Rating works but the endpoint needs to be implemented

### CODE REVIEW

File: src/main/java/com/example/tour/activity/ActivityController.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 @RestController
 @RequestMapping("api/activity")
 public class ActivityController {
-    private ActivityService activityService;
+    private final ActivityService activityService;
```
Comment (ivankras) [inline src/main/java/com/example/tour/activity/ActivityController.java:12]: Does this work, with the "final" on it?

File: src/main/java/com/example/tour/rating/Rating.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import lombok.*;
+
+import javax.persistence.*;
+import java.time.LocalDate;
+@NoArgsConstructor
+@EqualsAndHashCode
+@ToString
+@Getter
+@Setter
+@Data
```
Comment (ivankras) [inline src/main/java/com/example/tour/rating/Rating.java:13]: @Data is a convenient shortcut annotation that bundles the features of @ToString, @EqualsAndHashCode, @Getter / @Setter and @RequiredArgsConstructor together

https://projectlombok.org/features/Data

File: src/main/java/com/example/tour/TourApplication.java [hunk 3]
```diff
@@ -19,13 +21,16 @@ public static void main(String[] args) {
     }
 
     @Bean
-CommandLineRunner commandLineRunner(PlaceRepository placeRepository, ActivityRepository activityRepository){
+CommandLineRunner commandLineRunner(PlaceRepository placeRepository, ActivityRepository activityRepository, RatingRepository ratingRepository){
```
Comment (ivankras) [inline src/main/java/com/example/tour/TourApplication.java:24]: commandLineRunner generates new entries on the database each time we run the project. It'd be nice to have it commented and use it when necessary. For final testing purposes, we'll need to implement JUnit tests

File: src/main/java/com/example/tour/activity/Activity.java [hunk 4]
```diff
@@ -39,6 +45,16 @@
...
     private Place place;
+    @OneToMany(
+            fetch = FetchType.EAGER,
+            mappedBy = "activity",
+            orphanRemoval = true,
+            cascade = CascadeType.ALL
+            //cascade = {CascadeType.PERSIST,CascadeType.REMOVE}
+    )
+//    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true, mappedBy = "activity")
+//    @LazyCollection(LazyCollectionOption.FALSE)
```
Comment (ivankras) [inline src/main/java/com/example/tour/activity/Activity.java:56]: If these commented lines are not necessary, delete them

File: src/main/java/com/example/tour/rating/Rating.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            strategy = GenerationType.SEQUENCE,
+            generator = "activity_sequence"
+    )
+    @Column(name = "id", updatable = false)
+    private Long id;
+    private LocalDate date;
+    private int score;
+
+    @ManyToOne (
+            //cascade = CascadeType.ALL,
```
Comment (ivankras) [inline src/main/java/com/example/tour/rating/Rating.java:34]: Delete if not necessary

File: src/main/java/com/example/tour/rating/RatingController.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    @GetMapping
+    public List<RatingDTO> getAllRating(){
+        return TransformerUtils.createListRatingDTO(ratingService.getRatings());
+    }
+
+    @PostMapping
+    public void addNewRating(@RequestBody Activity activity, Rating rating) {
+        if (activity == null || activity.isActive()) return;
+        ratingService.addNewRating(rating);
+    }
```
Comment (ivankras) [inline src/main/java/com/example/tour/rating/RatingController.java:29]: (now I remembered you said this endpoint was not working yet)

This code takes the fields of an Activity from a POST request, a Rating from nobody knows where, and adds a rating to the db only passing the rating? I don't get it. Does it work? (ignore this last question)

**Review by ivankras — CHANGES_REQUESTED** [review]
There are things to review before merging this into develop. Please work on them (if you need any of these in develop before ending, let me know). I'll leave the PR as Work In Progress (WIP)

File: src/main/java/com/example/tour/activity/ActivityController.java (hunk 1 above)
Comment (raeyh) [reply src/main/java/com/example/tour/activity/ActivityController.java:12]: This works actually,
I think this is the recommended one,

**Review by ivankras — APPROVED** [review]

File: src/main/java/com/example/tour/rating/RatingController.java [hunk 7]
```diff
@@ -7,18 +7,18 @@
 import java.util.List;
 
 @RestController
-@RequestMapping("api/rating/")
 
+@RequestMapping("/activities/ratings")
```
Comment (ivankras) [inline src/main/java/com/example/tour/rating/RatingController.java:11]: This should be @RequestMapping("activity/rating"), shouldn't it?

**Review by ivankras — APPROVED** [review]

**Pull request merged**